### ISSUE TITLE
Feaure Request: docker swarm mode state of service during startup

### ISSUE THREAD

**Issue by patran** [moby/moby#32775]
A docker service (docker swarm mode) may not startup until the network is there, the volumes are in place, etc.

It would be nice to have an easy way to identify the missing pieces. For example, if the service is waiting on a missing network, secret, or a volume or a label.

**Comment by cpuguy83** [comment]
Is this sufficient?

```
$ docker service create -d=false --replicas=5 --name test nginx                                                      ) ✗git:(master
xe1htfptqmhcq1lzv972lsuez
overall progress: 0 out of 5 tasks
1/5: preparing [=================================>                 ]
2/5: preparing [=================================>                 ]
3/5: preparing [=================================>                 ]
4/5: preparing [=================================>                 ]
5/5: preparing [=================================>                 ]
```

This is basically a new synchronous mode for `docker service create` that will be in 17.05. We plan to make it the default soon.

**Comment by patran** [comment]
@cpuguy83, nice !!! A couple of question:

If the service is stuck at step 2, would it be possible to know what it is waiting for? A free node, a network to be created etc?  Currently, I would run down a check list manually, but it would be nice if docker swarm mode could provide that information: hey, this service is stuck at this step because there is a missing volume, etc.

Similar to the previous question but with adversarial situations in mind: if docker crashed during, say step 3, and restarted, would it be possible to obtain the status that the service was on step 3 and what docker is planning to do with the service?

I am thinking here that with thousands of services being started, it would be nice to get a summary of the state of those services and what may need to get done (create network, volume,  label, etc) to enable services to get fully started

**Comment by cpuguy83** [comment]
@patran In most (all?) cases anywhere it can't proceed you'll see an error. I'm not sure what that looks like for all case.... however if it's an issue with a specifically checked item... like for instance a source path for a bind mount, you'll see a `rejected` status.... then you can `docker service ps <name>` to see the status.

In terms of a summary UI for this for multiple services... I'm not sure where it would fit in the CLI.
You can specify multiple services on `docker service ps`

ping @aaronlehmann

**Comment by patran** [comment]
@cpuguy83 , I will check out the next docker release and see if I can produce specific scenarios. The "filter" features may prove to be sufficient.

**Comment by thaJeztah** [comment]
The `--detach` option has been released, and docker 17.10 will use `--detach=false` as default. I'll go ahead and close this issue, but feel free to continue the conversation
